src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -47,25 +58,38 @@ public void start(Map<String, String> settings) {
         } catch (ChannelSinkConnectorConfig.ConfigException e) {
             logger.error("Failed to create Ably client", e);
         }
-        ablyClient.connect(isSuspended -> {
-            synchronized(ChannelSinkTask.this){
-                suspended.set(isSuspended);
-                if (!isSuspended){
-                    processSuspendQueue();
-                }
-            }
-        });
+//        ablyClient.connect(isSuspended -> {
+//            synchronized(ChannelSinkTask.this){
+//                suspended.set(isSuspended);
+//                if (!isSuspended){
+//                    processSuspendQueue();
+//                }
+//            }
+//        });
+        logger.info("Starting Ably channel Sink task");
+        // start the Batch processing thread.
+        this.sinkRecords = new ConcurrentLinkedQueue<>();
+        this.batchProcessingThread = new BatchProcessingThread(this.sinkRecords, this.ablyClient);
+        this.executor = new BatchProcessingExecutor(Integer.parseInt(settings
+                .get(ChannelSinkConnectorConfig.BATCH_EXECUTION_THREAD_POOL_SIZE)));
+
+        this.executor.scheduleAtFixedRate(this.batchProcessingThread, 0,
```

Comment:
Will there actually be any parallelism here? If I understand correctly, this code will create a thread pool of the desired size, but we only have a single `Runnable` instance as a member of this class and we just call this method to schedule it's execution periodically once?

Reply:
Fixed.